### DISCUSSION TITLE
Permissions on API tokens

### DISCUSSION THREAD

**Discussion by daggr8** [anonaddy/anonaddy#260] [Feature Requests]
Hi,

Is it possible to add permissions on API tokens? Meaning that a specific token could only be used for eg GET or GET aliases?
The use case for this could be a backup script which only requires GET aliases and nothing else.

**Comment by willbrowningme** [comment]
Yes that is possible. The permissions (abilities / scopes)  for each token could be: read, write, delete.

I plan to migrate from [Laravel Passport](https://laravel.com/docs/8.x/passport) to [Laravel Sanctum](https://laravel.com/docs/8.x/sanctum) which should make this simpler.

**Reply by daggr8** [reply]
Thank you for the reply! Cool, looking forward to it.

**Comment by daggr8** [comment]
Also possibility to lock down on what IP/hostnames that are allowed for a specific token?

**Reply by willbrowningme** [reply]
That should be possible by extending Sanctum's middleware or adding new middleware.
